decidim-proposals/app/commands/decidim/proposals/publish_proposal.rb

```diff
@@ -42,6 +35,39 @@ def call
 
       private
 
+      # Prevent PaperTrail from creating a version
+      # in the proposal multi-step creation process
+      def publish_proposal
+        PaperTrail.request(enabled: false) do
+          @proposal.update published_at: Time.current
+        end
+      end
+
+      # Set the definitive proposal version with: autor, title and body.
+      # A newly created proposal will have only one version, instead of
```

Comment:
Can you elaborate also here? I don't understand this 😅 

By this I mean `publish_proposal`, `set_proposal_version` and `reset`.

Reply:
https://github.com/decidim/decidim/pull/4567#discussion_r238595925 . Regarding `set_proposal_version`, this is where I let PaperTrail to create a new version with: author, title and body, which will be the one that shows in the version control feature.

The author part is easy, as you only have to pass the user to `perform_action!`; but in order for the version to show title/body as NEW changes, they have to be indeed new for that version, so resetting the values is the option I choosed (the `reset` method stores the value, resets the attribute and returns the value to be assigned in the final version).

Reply:
Refactored: https://github.com/decidim/decidim/pull/4567/commits/1acc0fed68ff377eec99bb06416c773a272f925f